scripts/leapfrog/README.md

```diff
@@ -0,0 +1,152 @@
+# Leapfrog Documentation
+
+#### Overview
+A Leapfrog upgrade is a major upgrade that skips at least one release. Currently
+RPCO supports leapfrog upgrades from kilo to newton-14.1.
+
+#### Terms
+
+* [RPCO](https://github.com/rcbops/rpc-openstack): Rackspace Private Cloud powered by OpenStack
+* [OSA](https://github.com/openstack/openstack-ansible):  OpenStack Ansible
+* [OSA-OPS](https://github.com/openstack/openstack-ansible-ops):  OpenStack Operations
+* [Kilo](https://github.com/rcbops/rpc-openstack/tree/kilo): The RPCO release of OpenStack Kilo
+* [Newton-14.1](https://github.com/rcbops/rpc-openstack/tree/newton-14.1) The second RPCO release of OpenStack Newton. This version includes Leapfrog upgrade tools.
+
+## Executing a leapfrog upgrade
+
+The start point is a kilo deployment. It is essential to check the kilo deployment
+is healthy before starting the leapfrog process. Many tools are availble for this,
+for example tempest.
+
+The next step is to checkout the newton-14.1 branch of rpco. This will remove
+any uncommitted information from the rpco repo, so back that up if it is
+required.
+
+```
+root@kilo:/opt/rpc-openstack# git reset --hard
+git clean -df
+pushd openstack-ansible
+git reset --hard
+git clean -df
+popd
+git fetch origin
+git checkout newton-14.1
+git submodule update
+git status
+```
+
+It is crucial that the rpco repo and openstack-ansible submodule are both clean
+and at the appropriate SHAs for newton-14.1 at this stage. The [github
+newton-14.1 branch
+page](https://github.com/rcbops/rpc-openstack/tree/newton-14.1) shows both SHAs,
+the RPCO SHA is the latest commit, and the OSA SHA is shown as an annotation
+after @ on the openstack-ansible submodule.
+
+Database housekeeping should be performed at this stage.
+
+The next step is to execute the leapfrog upgrade script
+
+```
+root@kilo:/rpc-openstack# scripts/leapfrog/ubuntu14-leapfrog.sh
+```
+
+## Structure of the leapfrog process.
+<img align="right" src="docs/images/leapfrog_structure_diagram.png"/>
+The RPCO leapfrog scripts are a thin wrapper around OSA-OPS leapfrog tools. 
+
+For details please refer to the scripts themselves. Paths are omitted for
+brevity, scripts may not be in the root of the relevant repo.
+
+
+#### Pre Leap
+This step removes modifications to RPCO Kilo that aren't compatible with RPCO
+Newton. Currently this only contains an Ansible 1.9 compatibility workaround.
+
+#### Prep
+This step executes pre-flight checks, and prompts the user for confirmation. It
+also ensures that the databases are backed up. Backups are stored in /var/backup
+on the first galera node.
```

Comment:
node is ambiguous - is that the host or the container? I'm guessing that's the location in the container, which has a bind mount to the host. I think it may be better to refer to the host's location and specifically state that it's available there.